content/blog/2022-05-09-CML-runners-saving-models-2.md

```diff
@@ -0,0 +1,240 @@
+---
+title:
+  Training and saving models with CML on a dedicated AWS EC2 runner (part 2)
+date: 2022-05-09
+description:
+  Use CML to automatically retrain a model on a provisioned AWS EC2 instance and
+  export the model to a DVC remote storage on Google Drive.
+descriptionLong: |
+  We can use CML to automatically retrain models whenever data, model code,
+  or parameters change. In this guide we show how to create a pipeline that
+  provisions an AWS EC2 instance to retrain a model and save the output on
+  a regular basis. In this part 2 we cover how to export the model to a DVC remote on Google Drive.
+picture: 2022-05-09/saving-models-2-cover.jpeg
+# pictureComment: Some _Comment_ (supports _basic_ [Markdown](link))
+author: rob_dewit
+# commentsUrl: https://discuss.dvc.org/t/february-22-community-gems/1078
+tags:
+  - CML
+  - DVC
+  - Git
+  - Pipelines
+  - Self-hosted runners
+  - Cloud training
+  - AWS
+  - Google Drive
+---
+
+In [part 1 of this guide](https://dvc.org/blog/CML-runners-saving-models-1) we
+showed how you can use CML to provision an AWS EC2 instance to train your model
+before saving the model to our Git repository. In doing so, we allowed ourselves
+to terminate the training instance without losing our model altogether.
+
+This worked perfectly fine for the simple model we trained, but this approach is
+not optimal when dealing with larger models. GitHub starts warning you at 50MB
+files and simply
+[won't upload anything over 100MB](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github).
+[GitLab similarly limits](https://docs.gitlab.com/ee/user/gitlab_com/index.html#account-and-limit-settings)
+the size of files you can store in your repository. A beefy XGBoost model can
+easily exceed 100MB and a neural network can go up into the gigabytes.
+
+That means we cannot save these models directly to our repository. Luckily we
+can look towards another one of Iterative's open-source tools:
+[DVC](https://dvc.org). DVC includes a lot of features for managing machine
+learning projects, such as ML pipelines and experiment tracking. In this guide
+we will zoom in on just one of those features: data versioning.
```

Comment:
Data versioning is mentioned here, but it seems like the focus of the post is storing the model in a remote. Maybe this could be reworded to say we'll use the remote feature of DVC to store large models?